HEC.FDA.ViewModel/ImpactAreaScenario/Results/SpecificIASResultVM.cs

```diff
@@ -8,199 +8,197 @@
+    {
+      ImpactAreaScenarioResults results = scenarioResults.GetResults(impactAreaID);
+      _IASResult = results;
+      LoadThresholdData(results);
+
+      loadVMs(damCats, scenarioResults, impactAreaID);
+      CurrentResultVM = _damageWithUncertaintyVM;
+
+      IASName = iasName;
+      SelectedOutcome = DAMAGE;
+
+      Reports = _damageReports;
+      SelectedReport = DAMAGE_WITH_UNCERTAINTY;
+    }
+    #endregion
+
+    #region Methods
+    private void LoadThresholdData(ImpactAreaScenarioResults iasResult)
+    {
+      foreach (Threshold threshold in iasResult.PerformanceByThresholds.ListOfThresholds)
+      {
+        ThresholdRowItem row = new ThresholdRowItem(threshold.ThresholdID, threshold.ThresholdType, threshold.ThresholdValue);
+        Thresholds.Add(new ThresholdComboItem(row.GetThreshold()));
+      }
+
+      if (Thresholds.Count > 0)
+      {
+        SelectedThreshold = Thresholds.First();
+      }
+    }
+
+    private void loadVMs(List<string> damCats, ScenarioResults scenarioResults, int impactAreaID)
+    {
+      StudyPropertiesElement studyPropElem = StudyCache.GetStudyPropertiesElement();
+      double discountRate = studyPropElem.DiscountRate;
+      int period = studyPropElem.PeriodOfAnalysis;
+
+      _damageWithUncertaintyVM = new DamageWithUncertaintyVM(scenarioResults,impactAreaID);
+      _damageByDamageCategoryVM = new DamageByDamCatVM(_IASResult, damCats, discountRate, period);
+      _performanceAEPVM = new PerformanceAEPVM(scenarioResults, impactAreaID, Thresholds);
+      _performanceAEPVM.updateSelectedMetric(SelectedThreshold);
+      _performanceAssuranceOfThresholdVM = new PerformanceAssuranceOfThresholdVM(_IASResult, Thresholds);
+      _performanceAssuranceOfThresholdVM.updateSelectedMetric(SelectedThreshold);
+      _performanceLongTermRiskVM = new PerformanceLongTermRiskVM(_IASResult, Thresholds);
+      _performanceLongTermRiskVM.updateSelectedMetric(SelectedThreshold);
+    }
+
+    private void ThresholdChanged()
+    {
+      if (_currentResultVM is PerformanceVMBase currentResult)
+      {
+        currentResult.updateSelectedMetric(SelectedThreshold);
+      }
+    }
+
+    /// <summary>
+    /// This method swaps out the viewmodel for the content control in the view.
+    /// </summary>
+    private void ReportChanged()
+    {
+      switch (SelectedReport)
+      {
+        case DAMAGE_WITH_UNCERTAINTY:
+          {
             CurrentResultVM = _damageWithUncertaintyVM;
-            
-            IASName = iasName;
-            Outcomes = _outcomes;
-            SelectedOutcome = DAMAGE;
+            break;
+          }
+        case DAMAGE_BY_DAMCAT:
+          {
+            CurrentResultVM = _damageByDamageCategoryVM;
+            break;
+          }
+        case ANNUAL_EXC_PROB:
+          {
+            CurrentResultVM = _performanceAEPVM;
+            break;
+          }
+        case LONG_TERM_RISK:
```

Comment:
this should be labeled long-term exceedance probability - not long-term risk

Reply:
repaired